Source/WTF/wtf/Logger.h

```diff
@@ -240,14 +241,24 @@ class Logger : public ThreadSafeRefCounted<Logger> {
     template<typename... Arguments>
     inline void debugVerbose(WTFLogChannel& channel, const char* file, const char* function, int line, const Arguments&... arguments) const
     {
-        if (!willLog(channel, WTFLogLevel::Debug))
+        if (!willLog(channel, WTFLogLevel::Debug, arguments...))
             return;
 
         logVerbose(channel, WTFLogLevel::Debug, file, function, line, arguments...);
     }
 
-    inline bool willLog(const WTFLogChannel& channel, WTFLogLevel level) const
+    template<typename... Argument>
+    inline bool willLog(const WTFLogChannel& channel, WTFLogLevel level, const Argument&... arguments) const
     {
+        {
+            if (!observerLock().tryLock())
+                return false;
+
+            Locker locker { AdoptLock, observerLock() };
+            for (Observer& observer : observers())
+                observer.handleLogMessage(channel, level, { ConsoleLogValue<Argument>::toValue(arguments)... });
```

Comment:
We do not want to construct a Vector except if handleLogMessage is actually implemented.
Otherwise we create a Vector for no good reason.
Hence why a separate observer list might be better, or some boolean state telling whether to call handleLogMessage.